### PR TITLE
Implement display_details method for Book struct

### PR DESCRIPTION
**Pull request by unpeturbed**
- Add display_details method to Book struct
- Update main function to use new method

This change implements the requested functionality to display
book details using a method, rather than relying on the Debug trait.

### CODE REVIEW

File: 02-intermediate-rust/01_structs/structs_02/src/main.rs [hunk 1]
```diff
@@ -10,11 +10,20 @@ struct Book {
     pages: u32
 }
 
+//Implemtent a method for the book struct to display its details.
```
Comment (fnabinash) [inline 02-intermediate-rust/01_structs/structs_02/src/main.rs:13]: Can you please remove the problem statement?

Rule - The problem statement always be on top of the file.

**Review by fnabinash — CHANGES_REQUESTED** [review]
Your pr is perfect. Meets its requirements.

But a few things need to be adjusted.
- always put the problem statement on the first line. (it will be easy for newcomers.)
- Remove `#[derive(Debug)]` because we do not need that anymore.
- Remove `#[allow(dead_code)]` because we used all the fields of `struct` so we'll not get a warning.
- Remove the comment line `// To get rid of warnings, so annoying.`

But a huge **Thank You** for your time and efforts. 

Your name on https://github.com/fnabinash/rust-practice/discussions/17 soon as the first contributor. 😍😍

**Comment by unpeturbed** [comment]
Thank you, Abinash, All noted and corrected.

**Comment by fnabinash** [comment]
Congratulation 👍🏻🥳🚀[unpeturbed](https://github.com/unpeturbed).

We got our first contributor. keep doing, Keep improving.

**Pull request merged**